### PR TITLE
better arrowdodge, add simulationSteps

### PR DESCRIPTION
**Pull request by C10udburst**
- made arrowdoge use ProjectileEntitySimulator (also shouldn't jump into lava now i think)
- add a way to limit how many steps to calculate for each projectile

### CODE REVIEW

File: src/main/java/meteordevelopment/meteorclient/systems/modules/combat/ArrowDodge.java [hunk 1]
```diff
@@ -79,59 +94,53 @@
...
-            if (ignoreOwn.get() && ((ProjectileEntityAccessor) e).getOwnerUuid().equals(mc.player.getUuid())) continue;
-
-            List<Box> futureArrowHitboxes = new ArrayList<>();
-
-            for (int i = 0; i < arrowLookahead.get(); i++) {
-                Vec3d nextPos = e.getPos().add(e.getVelocity().multiply(i / 5.0f));
-                futureArrowHitboxes.add(new Box(
-                    nextPos.subtract(e.getBoundingBox().getXLength() / 2, 0, e.getBoundingBox().getZLength() / 2),
-                    nextPos.add(e.getBoundingBox().getXLength() / 2, e.getBoundingBox().getYLength(), e.getBoundingBox().getZLength() / 2)));
+        for (Vec3 point: points) vec3s.free(point);
```
Comment (MineGame159) [inline src/main/java/meteordevelopment/meteorclient/systems/modules/combat/ArrowDodge.java:107]: Space before :

File: src/main/java/meteordevelopment/meteorclient/systems/modules/combat/ArrowDodge.java [hunk 2]
```diff
@@ -79,59 +94,53 @@
...
+        points.clear();
+        mc.world.getEntities().forEach(e -> {
```
Comment (MineGame159) [inline src/main/java/meteordevelopment/meteorclient/systems/modules/combat/ArrowDodge.java:109]: Please use normal for each loop 😩

File: src/main/java/meteordevelopment/meteorclient/systems/modules/combat/ArrowDodge.java [hunk 3]
```diff
@@ -79,59 +94,53 @@
...
+            if (!(e instanceof ProjectileEntity)) return;
+            if (ignoreOwn.get() && ((ProjectileEntityAccessor) e).getOwnerUuid().equals(mc.player.getUuid())) return;
+            if (!simulator.set(e, accurate.get(), 0.5D)) return;
+            if (simulationSteps.get() == 0) {
+                while (true) {
+                    points.add(vec3s.get().set(simulator.pos));
+                    if (simulator.tick() != null) break;
+                }
+            } else {
+                for (int i = 0; i < simulationSteps.get(); i++) {
```
Comment (MineGame159) [inline src/main/java/meteordevelopment/meteorclient/systems/modules/combat/ArrowDodge.java:119]: You could merge the two loops by using `Integer.MAX_VALUE` as the limit if `simulationSteps` is 0

File: src/main/java/meteordevelopment/meteorclient/systems/modules/render/Trajectories.java [hunk 4]
```diff
@@ -192,16 +200,30 @@ public void clear() {
...
+    
+                    if (result != null) {
+                        processHitResult(result);
+                        break;
+                    }
+    
+                    addPoint();
+                }
+            } else {
+                for (int i = 0; i < simulationSteps.get(); i++) {
```
Comment (MineGame159) [inline src/main/java/meteordevelopment/meteorclient/systems/modules/render/Trajectories.java:215]: Same as before. You could merge the two loops by using `Integer.MAX_VALUE` as the limit if `simulationSteps` is 0.

**Review by MineGame159 — CHANGES_REQUESTED** [review]

File: src/main/java/meteordevelopment/meteorclient/systems/modules/combat/ArrowDodge.java (hunk 2 above)
Comment (C10udburst) [reply src/main/java/meteordevelopment/meteorclient/systems/modules/combat/ArrowDodge.java:109]: ok 😔

**Pull request merged**